hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/DirectMarkerTransactionManager.java

```diff
@@ -83,7 +83,7 @@ private static TypedProperties createUpdatedLockProps(
       throw new HoodieNotSupportedException("Only Support ZK-based lock for DirectMarkerTransactionManager now.");
     }
     TypedProperties props = new TypedProperties(writeConfig.getProps());
-    props.setProperty(LockConfiguration.ZK_LOCK_KEY_PROP_KEY, partitionPath + "/" + fileId);
+    props.setProperty(LockConfiguration.ZK_LOCK_KEY_PROP_KEY, (null != partitionPath && !partitionPath.isEmpty()) ? partitionPath + "/" + fileId : fileId);
```

Comment:
This change will pose a challenge when upgrading. During upgrade, we would need all writers to be stopped and upgrade to the version containing this change for safe concurrency

Reply:
On deeper look, this change should be fine as currently, the jobs are failing with exceptions when with the setup mentioned in the ticket.